docs/examples/intro_to_kernels.py

```diff
@@ -0,0 +1,328 @@
+#
+# The lengthscale, $\kappa$, and variance, $\sigma^2$, are shown below, both before and after optimisation:
+
+# %%
+no_opt_lengthscale = no_opt_posterior.prior.kernel.lengthscale
+no_opt_variance = no_opt_posterior.prior.kernel.variance
+opt_lengthscale = opt_posterior.prior.kernel.lengthscale
+opt_variance = opt_posterior.prior.kernel.variance
+
+print(f"Optimised Lengthscale: {opt_lengthscale} and Variance: {opt_variance}")
+print(
+    f"Non-Optimised Lengthscale: {no_opt_lengthscale} and Variance: {no_opt_variance}"
+)
+
+# %% [markdown]
+# ## Expressing Other Priors with Different Kernels
+
+# %% [markdown]
+# Whilst the Matérn kernels are often used as a first choice of kernel, and they often perform well due to their smoothing properties often being well-aligned with the properties of the underlying function being modelled, sometimes more prior knowledge is known about the function being modelled. For instance, it may be known that the function being modelled is *periodic*. In this case, a suitable kernel choice would be the *periodic* kernel:
+#
+# $$k(\mathbf{x}, \mathbf{x}') = \sigma^2 \exp \left( -\frac{1}{2} \sum_{i=1}^{D} \left(\frac{\sin (\pi (\mathbf{x}_i - \mathbf{x}_i')/p)}{\ell}\right)^2 \right)$$
+#
+# with $D$ being the dimensionality of the inputs.
+#
+# Below we show $10$ samples drawn from a GP prior using the periodic kernel:
+
+# %%
+mean = gpx.mean_functions.Zero()
+kernel = gpx.kernels.Periodic()
+prior = gpx.Prior(mean_function=mean, kernel=kernel)
+
+x = jnp.linspace(-3.0, 3.0, num=200).reshape(-1, 1)
+rv = prior(x)
+y = rv.sample(seed=key, sample_shape=(10,))
+
+fig, ax = plt.subplots()
+ax.plot(x, y.T, alpha=0.7)
+ax.set_title("Samples from the Periodic Kernel")
+plt.show()
+
+# %% [markdown]
+# In other scenarios, it may be known that the underlying function is *linear*, in which case the *linear* kernel would be a suitable choice:
+#
+# $$k(\mathbf{x}, \mathbf{x}') = \sigma^2 \mathbf{x}^\top \mathbf{x}'$$
+#
+# Unlike the kernels shown above, the linear kernel is *not* stationary, and so it is not invariant to translations in the input space.
+#
+# Below we show $10$ samples drawn from a GP prior using the linear kernel:
+
+# %%
+mean = gpx.mean_functions.Zero()
+kernel = gpx.kernels.Linear()
+prior = gpx.Prior(mean_function=mean, kernel=kernel)
+
+x = jnp.linspace(-3.0, 3.0, num=200).reshape(-1, 1)
+rv = prior(x)
+y = rv.sample(seed=key, sample_shape=(10,))
+
+fig, ax = plt.subplots()
+ax.plot(x, y.T, alpha=0.7)
+ax.set_title("Samples from the Linear Kernel")
+plt.show()
+
+# %% [markdown]
+# ## What are the Necessary Conditions for a Valid Kernel?
+
+# %% [markdown]
```

Comment:
It might be nice to expand our this section a little to lower the bar-to-entry e.g., define a Gram matrix and PSD, then move to the start of the notebook.

Reply:
Sure. This is your work and I'm happy for you to own the style. My comment was just a suggestion with the intention of making the notebook as accessible to new users as possible.